lib/components/form/LocationInput.tsx

```diff
@@ -1,262 +1,215 @@
-        newLat,
+        longitude,
+        latitude,
         canEdit
       );
 
-      setZoom(map.current!.getZoom());
-
       const isPastZoomThreshold = map.current!.getZoom() >= zoomLevelThreshold;
-
       if (isPastZoomThreshold) {
-        // save the map coordinates
-        saveSiteLocation();
+        setMapCoordinates({ longitude, latitude });
       }
-      setIsMoving(true);
-    });
+    };
 
-    map.current.on('moveend', () => {
+    const onMoveEnd = async () => {
       const isPastZoomThreshold = map.current!.getZoom() >= zoomLevelThreshold;
-      if (isPastZoomThreshold) {
-        // update the search box location based on the final latitude/longitude
-        geocoder.setFlyTo(false);
-
-        // don't display anything in suggestions
-        geocoder.setRenderFunction(() => '<div class="hidden-suggestion"/>');
-
-        // The below `_geocode` call performs the same action that the MapBox geocoder does, except it
-        // does not re-focus the input element after a successful geocode response
-        const geocoderPrivate = geocoder as any;
-        geocoderPrivate
-          ._geocode(`${savedLat}, ${savedLng}`)
-          .then((response: any) => {
-            const results = response.body;
-            if (!results.features.length) return;
-            const result = results.features[0];
-            geocoderPrivate._typeahead.selected = result;
-            geocoderPrivate._inputEl.value = result.place_name;
-            const selected = geocoderPrivate._typeahead.selected;
-            if (
-              selected &&
-              JSON.stringify(selected) !== geocoderPrivate.lastSelected
-            ) {
-              geocoderPrivate._hideClearButton();
-              if (geocoderPrivate.options.flyTo) {
-                geocoderPrivate._fly(selected);
-              }
-              if (geocoderPrivate.options.marker && geocoderPrivate._mapboxgl) {
-                geocoderPrivate._handleMarker(selected);
-              }
-
-              // After selecting a feature, re-focus the textarea and set
-              // cursor at start.
-              geocoderPrivate._inputEl.scrollLeft = 0;
-              geocoderPrivate.lastSelected = JSON.stringify(selected);
-              geocoderPrivate._eventEmitter.emit('result', {
-                result: selected,
-              });
-              geocoderPrivate.eventManager.select(selected, geocoder as any);
-            }
-          });
-
-        if (popup === null && map.current && canEdit) {
-          popup = new mapboxgl.Popup({
-            offset: [0, 10],
-            anchor: 'top',
-            closeOnClick: false,
-            closeButton: false,
-            className: 'site-map',
-          })
-            .setLngLat([savedLng, savedLat])
-            .setHTML('<p>Drag map to pinpoint exact location.</p>')
-            .addTo(map.current);
-        }
-      }
-      setIsMoving(false);
-      setIsPastZoomThreshold(isPastZoomThreshold);
```

Comment:
consider bringing this behavior back (where submission is disabled while moving or while not past zoom threshold) since this resolved some bugs in the past

Reply:
From some quick testing I haven't been getting the issue. Since there are now `off` calls to turn off event listeners once the `LocationInput` unmounts, these should be fixed.